File: adsterox123-cs50-2017-x-greedy/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>


int main(void)
{
    float amount;
    do
    {
        printf("O hai! How much change is owed?\n");
        amount = get_float();
    }
    while(amount < 0);
    
    amount *= 100;
    amount = (int) round(amount);
    int coins = 0;
    
    while(amount >= 25)
    {
        coins += 1;
        amount -= 25;
    }
    while(amount >= 10)
    {
        coins += 1;
        amount -= 10;
    }
    while(amount >= 5)
    {
        coins += 1;
        amount -= 5;
    }
    while(amount >= 1)
    {
        coins += 1;
        amount -= 1;
    }
    printf("%i\n", coins);
    
}

File: adsterox123-cs50-2017-x-pset6/greedy.py
while True:
    amount = float(input("How much change is owed "))
    if (amount >= 0):
        break


amount = amount * 100;
amount = round(amount, 0);
coins = 0;

while amount >= 25:

    coins = coins + 1;
    amount = amount - 25;

while amount >= 10:

    coins = coins + 1;
    amount = amount - 10;

while amount >= 5:

    coins = coins + 1;
    amount = amount - 5;

while amount >= 1:

    coins = coins + 1;
    amount = amount - 1;

print(coins)